sympy/polys/matrices/sdm.py

```diff
@@ -28,6 +28,24 @@ def __init__(self, elemsdict, shape, domain):
         if not all(0 <= c < n for row in self.values() for c in row):
             raise DDMBadInputError("Column out of range")
 
+    def getitem(self, i, j):
+        try:
+            return self[i][j]
+        except KeyError:
+            return self.domain.zero
```

Comment:
Another case is needed here. If we get `KeyError` then it should be checked whether `i` and `j` are valid indices and `IndexError` should potentially be raised.

Reply:
In the commit, we check for the indices in the `DomainMatrix` itself before calling `getitem`.
After the above discussion, we will now shift the checks from `DomainMatrix.__getitem__` to `SDM.getitem` for the case of single item accessing.
Thanks.